apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/Vote.java

```diff
@@ -60,16 +63,51 @@ public void trim(int height) {
             try (Connection con = databaseManager.getDataSource().getConnection();
                  DbIterator<Poll> polls = Poll.getPollsFinishingAtOrBefore(height, 0, Integer.MAX_VALUE);
                  PreparedStatement pstmt = con.prepareStatement("DELETE FROM vote WHERE poll_id = ?")) {
-                for (Poll poll : polls) {
-                    pstmt.setLong(1, poll.getId());
-                    pstmt.executeUpdate();
-                }
+                commonTrim(height, false, polls, pstmt);
```

Comment:
Did you consider a possibility to just decrease 'Poll.getPollsFinishingAtOrBefore' target height by one or just replace the whole method with new one you created (getPollsFinishingAt, i also suggested to rename it above) instead of supporting two trim variants and increase the complexity of derived tables hierarchy?

Reply:
I wouldn't like to mutate height inside internal method again then it was already mutated before|above. It makes logic not obvious, so I prefer do not do that here.